common/scala/src/main/scala/org/apache/openwhisk/core/WhiskConfig.scala

```diff
@@ -54,8 +54,16 @@ class WhiskConfig(requiredProperties: Map[String, String],
   val dockerEndpoint = this(WhiskConfig.dockerEndpoint)
   val dockerPort = this(WhiskConfig.dockerPort)
 
-  val wskApiHost = this(WhiskConfig.wskApiProtocol) + "://" + this(WhiskConfig.wskApiHostname) + ":" + this(
-    WhiskConfig.wskApiPort)
+  val wskApiHost = {
```

Comment:
Can possibly be simplified by using `akka.http.scaladsl.model.Uri.normalize`

```suggestion
  val wskApiHost = Uri.normalize(s"${this(WhiskConfig.wskApiProtocol)}://${this(WhiskConfig.wskApiHostname)}:${this(WhiskConfig.wskApiPort)}")
```

Reply:
awesome - thanks for the tip... is there a java equivalent to replace the changes in whisk props below?

Reply:
Do not know of any java equivalent and hard to invoke the Scala code from java. So lets retain the current approach for test code

Reply:
the change causes many tests to fail - it appears in a number of unit tests, the value of the string is just `://:` - and I think to guard the values i might end up with more code than the original version so im inclined to just use the explicit removal of the port. wdyt.

Reply:
Yeah seems like values are empty which cause the parser to fail. May be an approach like below?

```
  val wskApiHost = Try(
    Uri.normalize(
      this(WhiskConfig.wskApiProtocol) + "://" + this(WhiskConfig.wskApiHostname) + ":" + this(WhiskConfig.wskApiPort)))
    .getOrElse("")
```